### DISCUSSION TITLE
Remove unused JavaScript in auto generated .nuxt directory and files in node_modules

### DISCUSSION THREAD

**Discussion by larsdouweschuitema** [nuxt/nuxt#8849] [Questions (Nuxt 2)]
## Context

I decided to build a [Nuxt boilerplate](https://github.com/larsdouweschuitema-nuxt/nuxt-boilerplate) for myself that I can use as a template repository to create nuxt projects

The Google Lighthouse  performance is a 100 but still shows that I can save 0.15s (25.8 KiB) See attachments below:

### Google Lighthouse report

[score-perfect.pdf](https://github.com/nuxt/nuxt.js/files/5997683/score-perfect.pdf )

### Webpack analyzer bundle

<img src="https://user-images.githubusercontent.com/8817968/108247423-eea54e80-7152-11eb-9422-98a1b0fd2d29.png" width="200">

The .nuxt directory contains some auto generated files that are unused, and some of the dependencies nuxt depends on have unused files in them according to the webpack analyzer.

## Question 

How do I remove these node_modules and .nuxt auto generated files from the bundle?

## Attempt using .nuxtignore

Tried an attempt using .nuxtignore but that did not work out as expected. The files are still in the .nuxt generated folder.

```
./node_modules/deepmerge/dist/cjs.js
./node_modules/vue-style-loader/lib/addStylesClient.js
./node_modules/@nuxt/components/lib/installComponents.js
./node_modules/css-loader/dist/runtime/api.js
./.nuxt/pwa/meta.json
./.nuxt/components/nuxt-error.vue
```

**Comment by sdoerger** [comment]
I am looking at the same problem. Did you find a solution?

**Reply by larsdouweschuitema** [reply]
I think the solution would require an update to the core library so that it will exclude unused "components" and "libraries".
A similar option as disabling the core features would be nice so that bundle size is reduced.

**Reply by sdoerger** [reply]
Hmm, ok. Thanks for your answer.
Anything, that could be done, to please google lighthouse ... or is it something to just acknowledge?

Thanks again.

**Reply by amrshakya** [reply]
Is there any suggestion i am also looking for the solution for this but couldn't find anything yet. Any help is really appreciated.

**Reply by dvago** [reply]
Same here, it would be really appreciated to find a clean solution for this @Atinux

**Reply by mcmiro** [reply]
I have the same problem. :)

**Reply by MOHAMMADHMEAD** [reply]
any solution ?

**Reply by JenuelDev** [reply]
I am also searching for a solution:
![image](https://user-images.githubusercontent.com/31676163/235546342-4e4464c0-a7ec-421e-b676-8441fb8ebbdf.png)

**Comment by zakarialounes** [comment]
Also a problem on nuxt3.

**Reply by notflip** [reply]
I just came back to this thread, since I'm running into these issues again. Has anyone been able to come up with a solution?

**Reply by zweeen** [reply]
Any news about this opportunity to improve ? 😀

**Reply by hoachnt** [reply]
Nothing changed man
